analyze_facebook_data.rb

```diff
@@ -258,103 +303,153 @@ def most_popular_english_words
   end
 end
 
-# analyze making of friends
-friends_dates = []
-Dir.chdir("#{catalog}/html/") do
-  content = File.open('friends.htm').read
-  doc = Nokogiri::HTML(content)
-  friends_list = doc.css('div.contents > ul')[0].css('li')
+class FriendsDates
```

Comment:
I would like to keep the code in one place, in order to keep the overview (at least then I can see the patterns better, to complete the refactor) - for now.

In general agree with you, but I would make this the last step of refactoring.

What I would do next:
- Extract the creation of the excel sheet
- extract the `conversation.each` into a class
- probably want to make this into a GEM; adding folder structure for that.
- add rspec testing

Reply:
Would recommend to avoid trying to do a giant PR. This is my recommendation for the next set of PRs.

1. Refactor base code (this one)
2. Refactor second pass (separate in modules)
3. Testing (rspec + fake export)
4. Gemify the script

I can help out with test and the gemification.